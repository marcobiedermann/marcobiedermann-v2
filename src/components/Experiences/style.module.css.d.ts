/* eslint-disable import/prefer-default-export */

export const experiences: string;
