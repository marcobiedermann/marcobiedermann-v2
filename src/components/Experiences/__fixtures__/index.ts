import experience from '../../Experience/__fixtures__';

const experiences = [experience];

export default experiences;
