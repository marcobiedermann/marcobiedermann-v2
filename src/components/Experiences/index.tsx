import classNames from 'classnames';
import React from 'react';
import Experience, { ExperienceProps } from '../Experience';
import styles from './style.module.css';

export interface ExperiencesProps {
  className?: string;
  experiences: ExperienceProps[];
}

const Experiences: React.FC<ExperiencesProps> = props => {
  const { className, experiences } = props;

  return (
    <ul className={classNames(className, styles.experiences)}>
      {experiences.map(experience => (
        <li key={experience.id}>
          <Experience {...experience} />
        </li>
      ))}
    </ul>
  );
};

export default Experiences;
