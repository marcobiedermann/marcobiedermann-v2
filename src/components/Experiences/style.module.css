.experiences {
  list-style: none;
  margin-block: 1.5em;
  margin-top: 0.5em;
  padding: 0;
}
