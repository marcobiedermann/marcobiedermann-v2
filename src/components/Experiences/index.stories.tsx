/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Experiences from '.';
import experiences from './__fixtures__';

storiesOf('Experiences', module).add('default', () => <Experiences experiences={experiences} />);
