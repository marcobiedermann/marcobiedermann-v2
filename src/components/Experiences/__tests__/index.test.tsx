/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Experiences from '..';
import experiences from '../__fixtures__';

describe('Experiences component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Experiences experiences={experiences} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
