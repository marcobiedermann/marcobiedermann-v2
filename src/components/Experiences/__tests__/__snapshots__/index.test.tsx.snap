// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiences component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class=""
  >
    <li>
      <div>
        <h3>
          Job title
        </h3>
        <h4>
          @
          <a
            href="http://www.example.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Company name
          </a>
          , Berlin, DE
        </h4>
        <div>
          Jan 2000 â€“ Jan 2001 (1 year)
        </div>
        <div />
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
